pkg/client/dtclient/config_client.go

```diff
@@ -35,7 +35,36 @@ import (
 	"strings"
 )
 
-func (d *DynatraceClient) upsertDynatraceObject(ctx context.Context, theApi api.API, objectName string, payload []byte) (DynatraceEntity, error) {
+// APIData holds information about the api of the config to be read or written on the server
+type APIData struct {
```

Comment:
The way I had originally understood it, adding a different struct would allow to hold differing data to the api.API struct / allow removing that need to hold parent entity ID in the api.API struct. 

With the current change you've added the parent ID to the value "wrapper" specific to download (which looks like the perfect place).

This leaves this APIData struct as more or less a direct slightly slimmed down version of api.API - am I missing something, or do we maybe currently not need this struct after all?